### PR TITLE
Stabilize ignore config option

### LINKED ISSUE #3243: Stabilize the `ignore` configuration
This one does not affect how the code is formatted but allows specific files or directories to be excluded from formatting.

What is needed for stabilizing this for one of the next releases?

### PR DESCRIPTION
**Pull request by topecongiro**
Close #3243.

### CODE REVIEW

File: Configurations.md [hunk 1]
```diff
@@ -1063,7 +1063,7 @@ The pattern format is the same as [.gitignore](https://git-scm.com/docs/gitignor
 
 - **Default value**: format every file
 - **Possible values**: See an example below
-- **Stable**: No (tracking issue: [#3395](https://github.com/rust-lang/rustfmt/issues/3395))
+- **Stable**: Yes
```
Comment (calebcartwright) [inline Configurations.md:1066]: Do you think we should fix #3881 before stabilizing this? Or shall we go ahead and stabilize but treat #3881 as bug?

**Comment by calebcartwright** [comment]
CI failures are spurious, presumably related to the current [GH status issues](https://www.githubstatus.com/)

```bash
error: failed to get `ansi_term` as a dependency of package `rustfmt-bin v2.0.0-rc.1 (/Users/travis/build/rust-lang/rustfmt/rustfmt-core/rustfmt-bin)`

Caused by:

  failed to load source for dependency `ansi_term`

Caused by:

  Unable to update registry `https://github.com/rust-lang/crates.io-index`

Caused by:

  failed to fetch `https://github.com/rust-lang/crates.io-index`

```

File: Configurations.md (hunk 1 above)
Comment (topecongiro) [reply Configurations.md:1066]: Good catch. I will fix that first. Thanks!

Comment (topecongiro) [reply Configurations.md:1066]: #3881 is fixed by #4179.

**Comment by kpcyrd** [comment]
Should this work by now?
```
% cargo --version
cargo 1.52.0 (69767412a 2021-04-21)
% rustfmt --version
rustfmt 1.4.36-stable (7de6968 2021-02-07)
% cat .rustfmt.toml 
ignore = [
    "src/schema.rs"
]
% cargo fmt
Warning: can't set `ignore = IgnoreList { path_set: {"src/schema.rs"}, rustfmt_toml_path: "" }`, unstable features are only available in nightly channel.
Warning: can't set `ignore = IgnoreList { path_set: {"src/schema.rs"}, rustfmt_toml_path: "" }`, unstable features are only available in nightly channel.
% 
```

**Comment by misha** [comment]
Can confirm the same error with a slightly older version, too.

```
cargo 1.50.0 (f04e7fab7 2021-02-04)
$ rustfmt -V
rustfmt 1.4.30-stable (8c6769d 2021-01-18)
$ cat rustfmt.toml 
ignore = [
    "src/kbn_constants.rs",
]
$ cargo fmt
Warning: can't set `ignore = IgnoreList { path_set: {"src/kbn_constants.rs"}, rustfmt_toml_path: "" }`, unstable features are only available in nightly channel.
```

**Comment by cloutiertyler** [comment]
Same here:

```
$ cargo -V
cargo 1.52.0 (69767412a 2021-04-21)
$ rustfmt -V
rustfmt 1.4.36-stable (7de6968e 2021-02-07)
$ cat .rustfmt.toml
ignore = [
    "src/messages/test.rs",
]
$ cargo fmt
Warning: can't set `ignore = IgnoreList { path_set: {"src/messages/test.rs"}, rustfmt_toml_path: "" }`, unstable features are only available in nightly channel.
```

**Comment by nlhepler** [comment]
Was this commit excised from master after commit? I cannot find it in the master branch history.

**Comment by ahirner** [comment]
Issue persisted in:
```
$ cargo -V
cargo 1.56.0 (4ed5d137b 2021-10-04)
$ rustfmt -V
rustfmt 1.4.37-stable (09c42c45 2021-10-18)
```

**Comment by calebcartwright** [comment]
All - your definitive source of truth around which rustfmt configuration options are available, as well as which are stable/unstable, in the version of rustfmt _you_ are using is the configuration website: https://rust-lang.github.io/rustfmt/?version=v1.4.38&search=#ignore

Changes made to source control are not reflective of any and all released versions of software, especially in multibranch repositories, and the `ignore` option is still unstable in all released versions of rustfmt.

This change was originally made against an exploratory 2.0 version of rustfmt along with a multitude of other breaking changes. That 2.0 version of rustfmt is unlikely to happen (at least any time soon), and as such that version is no longer being tracked on the default/master branch of this repository which is why the commits from this, and other 2.0-bound PRs do not exist on the current master branch, refs #4801. It also means that there are merged PRs and associated closed issues which haven't actually made their way into a release line.

There's a large number of such changes that were made for that potential 2.0 release which we have to revisit individually, including this one, and I'd ask that folks refrain from continuing to post the "me too" types of comments; it's known, and the issues are being tracked.

In the interim, I would again encourage always referencing the config site and check the actual [release notes](https://github.com/rust-lang/rustfmt/blob/master/CHANGELOG.md) for details on what's indeed been released.

**Comment by rdaum** [comment]
This remains broken in 2024.

``` 
Warning: can't set `ignore = IgnoreList { path_set: {"submodules"}, rustfmt_toml_path: "" }`, unstable features are only available in nightly channel.
...
rustfmt 1.7.0-stable (07dca48 2024-02-04)
...
active toolchain
----------------

stable-x86_64-unknown-linux-gnu (default)
rustc 1.76.0 (07dca489a 2024-02-04)
```

**Comment by ytmimi** [comment]
@rdaum please see https://github.com/rust-lang/rustfmt/issues/3243#issuecomment-1954590357

**Pull request merged**